server/network/aoprotocol.py

```diff
@@ -375,224 +380,126 @@ def net_cmd_cc(self, args):
-                                 ):
-            # Pre-2.6 validation monstrosity.
-            msg_type, pre, folder, anim, text, pos, sfx, anim_type, cid, sfx_delay, button, evidence, flip, ding, color = args
-        elif self.validate_net_cmd(
-            args, self.ArgType.STR, self.ArgType.STR_OR_EMPTY,              # msg_type, pre
-            self.ArgType.STR, self.ArgType.STR, self.ArgType.STR_OR_EMPTY,  # folder, anim, text
-            self.ArgType.STR, self.ArgType.STR, self.ArgType.INT,           # pos, sfx, anim_type
-            self.ArgType.INT, self.ArgType.INT, self.ArgType.INT_OR_STR,    # cid, sfx_delay, button
-            self.ArgType.INT, self.ArgType.INT, self.ArgType.INT,           # evidence, flip, ding
-            # color, showname, charid_pair
-            self.ArgType.INT, self.ArgType.STR_OR_EMPTY, self.ArgType.INT,
-            # offset_pair, nonint_pre
-            self.ArgType.STR, self.ArgType.INT,
-        ):
-            # 2.6 validation monstrosity.
-            msg_type, pre, folder, anim, text, pos, sfx, anim_type, cid, sfx_delay, button, evidence, flip, ding, color, showname, charid_pair, offset_pair, nonint_pre = args
-        elif self.validate_net_cmd(
-            args, self.ArgType.STR, self.ArgType.STR_OR_EMPTY,              # msg_type, pre
-            self.ArgType.STR, self.ArgType.STR, self.ArgType.STR_OR_EMPTY,  # folder, anim, text
-            self.ArgType.STR, self.ArgType.STR, self.ArgType.INT,           # pos, sfx, anim_type
-            self.ArgType.INT, self.ArgType.INT, self.ArgType.INT_OR_STR,    # cid, sfx_delay, button
-            self.ArgType.INT, self.ArgType.INT, self.ArgType.INT,           # evidence, flip, ding
-            # color, showname, charid_pair
-            self.ArgType.INT, self.ArgType.STR_OR_EMPTY, self.ArgType.STR,
-            # offset_pair, nonint_pre, sfx_looping
-            self.ArgType.STR, self.ArgType.INT, self.ArgType.STR,
-            # screenshake, frames_shake, frames_realization
-            self.ArgType.INT, self.ArgType.STR, self.ArgType.STR,
-            # frames_sfx, additive, effect
-            self.ArgType.STR, self.ArgType.INT, self.ArgType.STR,
-        ):
-            # 2.8 validation monstrosity. (rip 2.7)
-            msg_type, pre, folder, anim, text, pos, sfx, anim_type, cid, sfx_delay, button, evidence, flip, ding, color, showname, charid_pair, offset_pair, nonint_pre, sfx_looping, screenshake, frames_shake, frames_realization, frames_sfx, additive, effect = args
-            pair_args = charid_pair.split("^")
-            charid_pair = int(pair_args[0])
-            if (len(pair_args) > 1):
-                pair_order = pair_args[1]
-        else:
+        if self.client.area.can_send_message(self.client) is False:
             return
 
-        if len(showname) > 0 and not self.client.area.showname_changes_allowed:
+        packet_28 = MS_28.from_args(args)
+
+        if self.client.area.showname_changes_allowed is False:
```

Comment:
Why did you change all this stuff to `if x is False` instead of `if not x`. It's semantically not the same thing.

Reply:
I got a little to ambitious in making the code read a little better.

I didn't take into consideration that some of the properties might not be returning a Singleton. That's my bad.